tests/software_tests/can/addressing/test_normal_addressing.py

```
@@ -0,0 +1,569 @@
+        mock_is_compatible_can_id.return_value = True
+        with pytest.raises(NotImplementedError):
+            NormalFixedCanAddressingInformation.decode_can_id_ai_params(can_id=can_id)
+        mock_validate_can_id.assert_called_once_with(can_id)
+        mock_is_compatible_can_id.assert_called_once_with(can_id)
+
+    @pytest.mark.parametrize("can_id", [0xFFFFFFFFFFFF, "something"])
+    @patch(f"{SCRIPT_LOCATION}.NormalFixedCanAddressingInformation.is_compatible_can_id")
+    @patch(f"{SCRIPT_LOCATION}.CanIdHandler.validate_can_id")
+    def test_decode_can_id__value_error(self, mock_validate_can_id, mock_is_compatible_can_id, can_id):
+        mock_is_compatible_can_id.return_value = False
+        with pytest.raises(ValueError):
```

Ruff (0.12.2)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)